Cpp_hands_on_programming/lib/pipe/Ipc_pipe.cc

```diff
@@ -0,0 +1,89 @@
+#include "Ipc_pipe.h"
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <limits.h>
+
+void Ipc_pipe::send() {
+    std::cout << "Starting pipeSend.." << std::endl;
+	#if 1
+	// remove potentially existing FIFO
+	remove(name.c_str());    
+	int fdp, size, total(0), bytes_read(0), bytes_write(0);
+	std::vector<char> buffer(PIPE_BUF);
+	std::ifstream in;
+
+	// create a name pipe
+	if (mkfifo(name.c_str(), 0666) == -1) {
+		perror("mkfifo");
+		exit(EXIT_FAILURE);
+	}
+
+	// open name pipe
+	fdp = open(name.c_str(), O_WRONLY);
+	if (fdp == -1) {
+		throw(std::runtime_error("Ipc_pipe::send: open pipe"));
```

Comment:
Here, you're throwing exception but few lines above - 'exit'. Why is that?

Reply:
I missed that during refactoring, will fix shortly